src/agent/src/device.rs

```diff
@@ -911,6 +912,76 @@ async fn vfio_ap_device_handler(_: &Device, _: &Arc<Mutex<Sandbox>>) -> Result<S
     Err(anyhow!("AP is only supported on s390x"))
 }
 
+
+#[instrument]
+pub async fn handle_cdi_devices(
+    devices: &[Device],
+    spec: &mut Spec,
+    sandbox: &Arc<Mutex<Sandbox>>,
+) -> Result<()> {
+        // If we have a sandbox container we do not inject any devices, the 
+        // CDI devices are used to create the proper PCIe topology and in case
+        // of cold-plug to direct-attach them to the VM.
+        if "pod_sandbox" == spec.annotations["io.katacontainers.pkg.oci.container_type"] {
+            return Ok(());
+        }
+
+        let raw_oci_config = Path::new("/tmp/raw-oci-config.json");
+        let cdi_oci_config = Path::new("/tmp/cdi-oci-config.json");
+
+        spec.save(raw_oci_config.to_str().unwrap())?;         
+ 
+        if !which("/usr/bin/cdi").is_ok() {
+            error!(sl(), "cdi not found in path");
+            return Ok(());
+        }
+
+        match  tokio::process::Command::new("/usr/bin/cdi")
```

Comment:
This needs to be replaced with proper Rust implementation of CDI. For now using a go binary to do the proper injection of devices. Until then this PR remains a draft.

Reply:
Hi Zvonko, does just execute cmd `cdi raw_oci_config cdi_oci_config` enough ? there's no such usage of cdi when I try to execute /usr/bin/cdi. Is it some specified version of cdi ?

Reply:
This is a stripped down version of cdi but we could use the official one.